weblate/checks/format.py

```diff
@@ -449,64 +453,64 @@ class PythonFormatCheck(BasePrintfCheck):
 
     check_id = "python_format"
     name = _("Python format")
-    description = _("Python format string does not match source")
+    description = _("Python-format string does not match source")
```

Comment:
Alternatively "Mismatching PHP-format string" ? (for this and others)

Reply:
Why "Python-format" instead of "Python format"?

Reply:
Because it isn't a "Python format-string".

Reply:
Everybody seems to call that "Python format string". See for example https://www.gnu.org/software/gettext/manual/html_node/python_002dformat.html or https://podalirius.net/en/articles/python-format-string-vulnerabilities/.

Reply:
http://www.grammar-monster.com/lessons/hyphens_in_compound_adjectives.htm
Word-salad like "Python brace format strings" and "Python % format strings" can be avoided.
The passage doesn't use em-dash or correct quoting either. Italics would be an option.

I didn't know about it either till some guy came and complained about it here, and we had the discussion.

Reply:
Thanks for this comment!

Yes, there are many places where hyphenated compound adjectives should be used in Weblate docs and are not. Yet. Repairing this will take a long time, and we are open to doing it. 
It just does not have the highest priority because:
- It has to be done very cautiously; in this case it is not as easy as putting `-` in front of every `format` word. Some formats’ names are a single word, some multiple words. The editor needs to know the topic and be careful; checking the documentation of that particular format can also be useful.
- I like grammar, I love rules! In Weblate docs, we do not always put grammar first, though. The majority of docs’ readers and writers do not know some grammar rules, even if they are native English speakers. Hyphenated-compound adjectives simply don’t bother many people; on the other hand, some readers are confused by it.
- Many projects (like file formats here) do not use correct grammar in their communities and docs. We simply reused bad practices, and it is hard to overcome it.

Next steps: Let’s do it. Systematically, together, without rush and spiking emotions. What do you think, @comradekingu?